### PR TITLE
React: Move to vitest

### LINKED ISSUE #3751: React: Move to vitest

### PR DESCRIPTION
**Pull request by qmonmert**
Fix #3751 

TODO:
- [x] https://www.npmjs.com/package/vitest-sonar-reporter

### CODE REVIEW

**Comment by qmonmert** [comment]
I detect a bug, I work on it

**Comment by pascalgrimaud** [comment]
@qmonmert : can you share and describe what you found plz?

**Comment by qmonmert** [comment]
when you edit a test file with error, `npm t` does not show error

**Comment by qmonmert** [comment]
a problem with `vitest-sonar-reporter` because without it works but no sonar report

**Comment by qmonmert** [comment]
@pascalgrimaud bug fixed

**Comment by pascalgrimaud** [comment]
good thanks

File: src/main/resources/generator/client/react/vitest.config.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    globals: true,
+    logHeapUsage: true,
+    minThreads: 1,
+    maxThreads: 2,
+    environment: 'jsdom',
+    cache: false,
+    include: ['src/test/javascript/spec/**/*.(spec|test).(ts|tsx)'],
+    exclude: ['node_modules', 'src/test/javascript/integration/**/*.spec.ts'],
+    coverage: {
+        clean: true,
```
Comment (pascalgrimaud) [inline src/main/resources/generator/client/react/vitest.config.ts:26]: Plz fix indent

**Pull request merged**